distribution/std-lib/Base/src/Main.enso

```diff
@@ -1,6 +1,46 @@
 import Base.List
-import Builtins
+import Base.Vector
+from Builtins import Number, Unit
 
 from Builtins export all
 
 from Base.List export Nil, Cons
+from Base.Vector export Vector
+
+## Represents a right-exclusive range of integer values.
+type Range
+    type Range start end
+
+    ## Applies a function to each element in the range.
+
+       > Example
+         To print all the numbers from 1 to 100 use:
+         1.upto 101 . each IO.println
+    each function =
+        it start end = if start == end then Unit else
+            function start
+            it start+1 end
+        it this.start this.end
+        Unit
```

Comment:
Isn't this `Unit` unnecessary, since the last call of `it` returns `Unit` and each not-last call returns the result of the next one, so the call to `it` will return `Unit` anyway. Or is it some performance consideration or just a code style to make it more clear that `Unit` is returned?

Reply:
It's for performance – the strangeness of our TCO system makes this preferable. Should be fixed when we have opt-in TCO.

Reply:
Maybe a comment could be added so that we remember to remove it then? Or is it expected to stay for a long time and the comment may cause more confusion than usefullness?